### PR TITLE
20139 - organize aws step_function

### PR DESCRIPTION
**Pull request by bhavaniravi**
Part of #20139

Move step_function_* operators and sensors to `step_function.py`

### CODE REVIEW

File: tests/deprecated_classes.py [hunk 1]
```diff
@@ -1518,6 +1518,16 @@
         'airflow.providers.amazon.aws.operators.s3_file_transform.S3FileTransformOperator',
         'airflow.operators.s3_file_transform_operator.S3FileTransformOperator',
     ),
+    (
+        'airflow.providers.amazon.aws.operators.step_function.StepFunctionStartExecutionOperator',
+        'airflow.providers.amazon.aws.operators.s3_file_transform_operator.'
```
Comment (eladkal) [inline tests/deprecated_classes.py]: this is a wrong file name

File: tests/deprecated_classes.py [hunk 2]
```diff
@@ -1518,6 +1518,16 @@
...
+        'StepFunctionStartExecutionOperator',
```
Comment (eladkal) [inline tests/deprecated_classes.py]: what is this extra entry?

File: tests/deprecated_classes.py [hunk 3]
```diff
@@ -1518,6 +1518,16 @@
...
+    ),
+    (
+        'airflow.providers.amazon.aws.operators.step_function.StepFunctionGetExecutionOutputOperator',
+        'airflow.providers.amazon.aws.operators.step_function_get_execution_output.'
+        'StepFunctionGetExecutionOutputOperator',
```
Comment (eladkal) [inline tests/deprecated_classes.py]: same question

**Review by eladkal — CHANGES_REQUESTED** [review]
You are also missing edits to the test files of the operators / sensors.

Sensor:
https://github.com/apache/airflow/blob/main/tests/providers/amazon/aws/sensors/test_step_function_execution.py
should also be renamed and fix imports.

Operators:
https://github.com/apache/airflow/blob/main/tests/providers/amazon/aws/operators/test_step_function_get_execution_output.py

https://github.com/apache/airflow/blob/main/tests/providers/amazon/aws/operators/test_step_function_start_execution.py

also since the classes are small we can have both of them in a single `test_step_function.py` and we can remove the two individual test files.

File: tests/deprecated_classes.py (hunk 2 above)
Comment (bhavaniravi) [reply tests/deprecated_classes.py]: It's how long string have been handled in this file

https://github.com/apache/airflow/blob/main/tests/deprecated_classes.py#L474

**Comment by eladkal** [comment]
@bhavaniravi could you please address also the comments on test files?

**Comment by bhavaniravi** [comment]
> @bhavaniravi could you please address also the comments on test files?

Yes on it.

**Review by eladkal — APPROVED** [review]
LGTM

**Pull request merged**